src/components/v5/shared/Stepper/Stepper.tsx

```diff
@@ -32,20 +32,21 @@ function Stepper<TKey extends React.Key>({
           },
         )}
       >
-        {items.map(({ key, content, isSkipped, isHidden, heading }, index) => {
-          const { decor, ...restHeading } = heading;
-          const isNextStepOptional = items[index + 1]?.isOptional;
-          const isNextStepSkipped = items[index + 1]?.isSkipped;
+        {items.map(
+          ({ key, content, isSkipped, isHidden, heading, iconName }, index) => {
```

Comment:
Wasn't `iconName` passed using `...restHeading`?

Reply:
no it wasn't, when I remove `iconName` from here in StepperButton component  iconName is undefined